pkg/auth/auth.go

```diff
@@ -214,24 +215,39 @@ func (u *UserAuth) isAllowed(verb string, itemsToCheck []resource) ([]Action, er
 	return rejectedActions, nil
 }
 
+func uniqVerbs(current []string, new []string) []string {
+	resMap := map[string]bool{}
+	for _, v := range current {
+		if !resMap[v] {
+			resMap[v] = true
+		}
+	}
+	for _, v := range new {
+		if !resMap[v] {
+			resMap[v] = true
+			current = append(current, v)
+		}
+	}
+	return current
+}
+
 func reduceActionsByVerb(actions []Action) []Action {
 	resMap := map[string]Action{}
-	res := []Action{}
 	for _, action := range actions {
 		req := fmt.Sprintf("%s/%s/%s", action.Namespace, action.APIVersion, action.Resource)
 		if _, ok := resMap[req]; ok {
 			// Element already exists
-			verbs := append(resMap[req].Verbs, action.Verbs...)
 			resMap[req] = Action{
 				APIVersion: action.APIVersion,
 				Resource:   action.Resource,
 				Namespace:  action.Namespace,
-				Verbs:      verbs,
+				Verbs:      uniqVerbs(resMap[req].Verbs, action.Verbs),
```

Comment:
Woah - so yes, here, it works, but the call to `uniqVerbs` is modifying `resMap[req].Verbs` to the desired value *and* setting it back. You could instead just do a call to `uniqVerbs` after you set `resMap[req]`, and it would work currently, though it'd be simpler to reason about if it was just a pure function.